tests/artifact_key_filter_test.py

```diff
@@ -0,0 +1,92 @@
+import pyshelf.artifact_key_filter as akfilter
+from tests.unit_test_base import UnitTestBase
+
+
+class ArtifactKeyFilterTest(UnitTestBase):
```

Comment:
Looks good. I wonder if we should add a test case where there are `boto.s3.prefix.Prefix` objects in the list? Since `bucket.list()` returns a `boto.s3.bucketlistresultset.BucketListResultSet` which if casted to a list it is a list of `boto.s3.key.Key` and `boto.s3.prefix.Prefix` objects (prefix is like a directory).  All of this should work with Prefix objects because they have the same properties as a Key but I wonder if we should test that? Or just do it in a functional test later when this is actually used? Up to you. Just thought of that scenario.